File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String stringaPiuCorta = strs[0];
        
        for (int i = 1; i < strs.length; i++) {
            if (strs[i].length() < stringaPiuCorta.length()) {
                stringaPiuCorta = strs[i];
            }
        }
        int cont = 0;
        while(cont != strs.length) {
            for (String str : strs) {
                if (str.startsWith(stringaPiuCorta)) 
                    cont++;
                else {
                    stringaPiuCorta = stringaPiuCorta.substring(0, stringaPiuCorta.length() - 1);
                    cont = 0;
                    break;
                }
            }
        }
        return stringaPiuCorta;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        common_prefix = ""
        strs = sorted(strs)
        first_str = strs[0]
        last_str = strs[-1]
        
        for i in range(min(len(first_str), len(last_str))):
            if first_str[i] != last_str[i]:
                return common_prefix
            common_prefix += first_str[i]
            
        return common_prefix
                